### ISSUE TITLE
Age ranges

### ISSUE THREAD

**Issue by gbaman** [emfcamp/Website#1209]
Age ranges as a string input turned out a bit of a mess, trusting people to actually enter something sensible...
![2022-05-31 17 56 29](https://user-images.githubusercontent.com/833193/171230937-4bea059f-fed2-456e-9bdd-b5ce8203a336.jpg)

So next time, we should aim ideally for age buckets. I would suggest
- All ages
- 5+
- 8+
- 12+
- 16+
- 18+

Ideally then on the filtering page, we have a slider allowing folks to see all workshops suit for kids for example aged 8+
Even after some cleanup, this is what we have ended up with. 

<img width="882" alt="image" src="https://user-images.githubusercontent.com/833193/171231366-72aaaf09-c391-41a8-8ba9-3e57aaa96917.png">

Finally, I am just thinking it might be worth enforcing folks choosing an option, given quite a lot were left unspecified.

**Comment by Jonty** [comment]
Note: It can't just be age+, some things are specifically aimed at people _under_ a certain age and we do not want older people showing up at them

**Comment by lukegb** [comment]
Should we swap to (minimum, maximum) age or something?

**Comment by russss** [comment]
In my opinion we're drastically over-thinking this and I worry that we're going to start putting people off submitting things if they have to provide the precise age range for every submission.

I believe this ticket specifically referred to youth workshops and was actually resolved in 738c5cbab2b3cdbf68b9026818b4df46132d5725 (committed the same day the ticket was filed) which set some precise age ranges.

We should not be asking for precise age ranges for talk submissions - the "family friendly" checkbox is fine there.

**Comment by russss** [comment]
Also I think the easy answer for non-youth workshops is simply to not allow people to filter by age range. It feels a bit unnecessary.

**Comment by Jonty** [comment]
AFAIK we're only discussing workshops as talks don't have this anyway, nor should they.

As I said above I'm talking about buckets not exact age ranges - probably no more than ~5 for exactly that off-putting reason. The existing list from that commit is _way_ too granular as it is, and it isn't correct anyway - we need options for lower than age 3.

I'll sort out a list that makes sense tomorrow.

**Comment by philpem** [comment]
Sorry for barging in, but I just saw this as I was browsing and with an attendee perspective ... if I see "family friendly" tacked onto an event, I'll probably read the description with the assumption it's something aimed at kids' and parents, just to figure out if it's solely for that group or something open to all but particularly recommended for that group.

"Toddlers and parents only" or "Young children and parents only" would be something I'd take as a hard rule.

Similarly I expect parents may appreciate "not recommended for children" in the schedule - although you really need to say why. Heavy tools, risk of solder splash, terrible dad jokes, or what have you.

I think clarity on "this age only" events in signage and the schedule would be appreciated, but that's only my POV.
I was tagging along with some costume performers who were attracted by the interactive art installations, and we had a couple of "uh, are we supposed to be here?" moments when we found ourselves surrounded by young kids and parents.

I shall leave you to your discussion, apologies if I've intruded!

**Comment by lukegb** [comment]
This field is used for two things:

* Workshops
* Youth workshops

It isn't used for talks - I need to doublecheck this and make sure that the combobox doesn't show up during finalisation, but that's it.

The initial concern was, as I interpreted it, that providing a free text field for people to put whatever they want in means that someone has to go and clean that up later and try to map it onto a smaller set of things. That's maybe fine?

https://github.com/emfcamp/Website/commit/738c5cbab2b3cdbf68b9026818b4df46132d5725 didn't (AFAIK) fix this because during the CfP process we still provide a free-text box, which if not cleaned up leads to what we have in the first screenshot.

**Comment by gbaman** [comment]
I am fully in support of buckets, although am not sure we had anything needing an upper limit beyond aged 3 or 4 last year? 

This has got me thinking about talks and workshops in general and suitability for young people. With how extremely busy the youth programme was last year, it would be ideal if we had more information on talks/workshops that would be suitable for young people beyond our (heavily limited capacity) youth programme.
Perhaps this is the wrong place to be discussing it, but some key information for parents to help make the decision themselves (as @philpem mentioned) if a session would at least be "safe" for kids to attend, or are specifically encouraged to attend could be very useful?

For both workshops and talks, perhaps checkboxes for "warnings", for example "Strong language", "Dangerous tools/equipment", "Explicit content" wouldn't be a bad idea. As a bonus, having a separate checkbox to encourage families/youth to attend would be a big bonus, for example "family friendly". 

We will still end up with sessions in the middle that don't have warnings, and aren't specifically encouraging families/young people to attend, but at least it should help families make better decisions for both sides.

**Comment by Jonty** [comment]
@gbaman I plan to gather this in the second phase of finalisation for the CFP, as yes we really want this for other workshops too.

**Comment by Jonty** [comment]
@lukegb Right, we need to split this into two sets of age ranges - one just shown on youth workshops and one shown on main workshops.

For youth we want the following ranges:
* Under 4
* 5+
* 8+
* 10+

That said, after talking to some of the content team I suggest that for the CFP submission we leave this as free-text, and we add a finalisation field that allows us to classify this into a range properly before publishing.

This is because people are frequently awful at understanding the range that they can actually take and it requires some back-and-forth with us to figure out what the suitable age range actually is for a workshop. I'd prefer that they are more descriptive in the submission to help us figure this out.

**Comment by gbaman** [comment]
I'm just adding the below screenshot in for reference for how things turned out in 2024 (before any cleaning up), more for next time thinking.

<img width="902" alt="image" src="https://github.com/emfcamp/Website/assets/833193/fbbdcef3-74bd-4dae-ae12-97f8e26e7516">

**Comment by gbaman** [comment]
Just an update, for 2024 in the (manual) cleanup, I decided to go for the following buckets.

- All ages
- 5+
- 8+
- 10+
- 12+
- 16+
- 18+
